### ISSUE TITLE
Add forking endpoint 

### ISSUE THREAD

**Issue by khoroshevskyi** [pepkit/pephub#121]
Add forking functionality

**Comment by khoroshevskyi** [comment]
I have two ideas how to do this: 
1) Simply copy paste project: 
```
@project.post("/fork/{fork_namespace}")
async def fork(
    project: str,
    fork_namespace: str,
    tag: str = DEFAULT_TAG,
    namespace_access_list: List[str] = Depends(get_namespace_access_list),
    proj: peppy.Project = Depends(get_project),
    agent: PEPDatabaseAgent = Depends(get_db),
):

    if fork_namespace in (namespace_access_list or []):
        agent.project.create(project=proj, namespace=fork_namespace, name=project, tag=tag)
    else:
        raise HTTPException(
            401, f"Unauthorized for using {fork_namespace}"
        )
```
With this option we are loosing metadata: submission date, update date... .

2) add fork option to pepdbagent. With this option we can still have keep metadata

@nleroy917 @nsheff  What do you think about this two options? First is mostly implemented

**Comment by nleroy917** [comment]
I would emulate what [github does](https://docs.github.com/en/rest/repos/forks?apiVersion=2022-11-28#create-a-fork).

I would also go with the first approach. So,

`POST` to `/projects/{namespace}/{project}/forks`, where **`namespace/project`** is the registry path of the **PEP you want to fork**. Then in the body:

```
// body
{
  namespace: nleroy917
}
```
And then only allow forking of private repos if authorization header is given:
```
// headers
{
  Authorization: Bearer <access_token>
}
```

**Comment by nleroy917** [comment]
This should be solved in the latest release

**Comment by nleroy917** [comment]
This is now available in the latest release (`0.7.x`)
